app/src/main/java/me/ivg2/foodapp/barcode/GraphicOverlay.java

```diff
@@ -167,16 +164,30 @@ protected void onDraw(Canvas canvas) {
             if ((mPreviewWidth != 0) && (mPreviewHeight != 0)) {
                 mWidthScaleFactor = (float) canvas.getWidth() / (float) mPreviewWidth;
                 mHeightScaleFactor = (float) canvas.getHeight() / (float) mPreviewHeight;
-
             }
-
-            for (Graphic graphic : mGraphics) {
-                graphic.draw(canvas);
-                //Fire the intent to go to Girum's add item thing
+            Paint paint = new Paint();
+            paint.setColor(Color.BLACK);
+            paint.setAlpha(100);
+            canvas.drawRect(0, canvas.getHeight() - 450, canvas.getWidth(), canvas.getHeight(), paint);               //bottom rect
+            canvas.drawRect(0, 0, canvas.getWidth(), 450, paint);                                           //top rect
+            canvas.drawRect(0, 450, 300, canvas.getHeight() - 450, paint);                              //left rect
+            canvas.drawRect(canvas.getWidth() - 300, 450, canvas.getWidth(), canvas.getHeight() - 450, paint);  //right rect
+            paint.setColor(Color.RED);
+            paint.setStrokeWidth(14);
+            float[] pointsArray =
+                    {300 - paint.getStrokeWidth() / 2, 450, 375, 450,                                                                               //top left
```

Comment:
Let's make sure the formatter gets run on this.  Also, do we know that this won't shit a brick on a device that isn't your test device? :)  How are these numbers calculated?

Reply:
Also, I'm not saying we should make this work on every device (that might be a waste of time) but it is a consideration!